propagation.go

```diff
@@ -259,6 +267,30 @@ func (p *binaryPropagator) Extract(abstractCarrier interface{}) (SpanContext, er
 	return ctx, nil
 }
 
+// Converts a comma separated key value pair list into a map
+// e.g. key1=value1, key2=value2, key3 = value3
+// is converted to map[string]string { "key1" : "value1",
+//                                     "key2" : "value2",
+//                                     "key3" : "value3" }
+func parseCommaSeparatedMap(value string) map[string]string {
+	baggage := make(map[string]string)
+	kvpairs := strings.Split(value, ",")
+	for _, kvpair := range kvpairs {
+		s, err := url.QueryUnescape(strings.TrimSpace(kvpair))
```

Comment:
I didn't do that because an error in unescaping will terminate the parsing of the entire string. In the current approach, only the specific key value pair is affected